cmd/cosign/main.go

```diff
@@ -47,22 +49,39 @@ func main() {
 	}
 
 	if err := root.Parse(os.Args[1:]); err != nil {
-		if *verbose {
-			fmt.Print("verbose!")
+		printErrAndExit(err)
+	}
+
+	if *outputFilename != "" {
+		out, err := os.Create(*outputFilename)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "error: %v\n", errors.Wrapf(err, "Error creating output file %s", *outputFilename))
+			os.Exit(1)
 		}
-		fmt.Fprintf(os.Stderr, "error: %v\n", err)
-		os.Exit(1)
+		stdout := os.Stdout
+		stderr := os.Stderr
+		defer func() {
+			os.Stdout = stdout
+			os.Stderr = stderr
+			out.Close()
+		}()
+		os.Stdout = out
+		os.Stderr = out
 	}
 
 	if *debug {
 		logs.Debug.SetOutput(os.Stderr)
 	}
 
 	if err := root.Run(context.Background()); err != nil {
-		if *verbose {
-			fmt.Print("verbose!")
-		}
-		fmt.Fprintf(os.Stderr, "error: %v\n", err)
-		os.Exit(1)
+		printErrAndExit(err)
+	}
+}
+
+func printErrAndExit(err error) {
+	if *verbose {
+		fmt.Print("verbose!")
```

Comment:
I think this can get deleted :) It was a copy/paste error IIRC.

Reply:
I didn't know if it was part of more elaborate plans. :smile: I'll remove in the next push and remove the flag as well.